appcelerator/titanium_mobile
# [TIMOB-25859] Android: Delay WebView Ti.App.fireEvent() to be fired after page load

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25859

**Summary:**
- Events fired from HTML are now queued while the page is loading and are fired when page finishes loading. This is needed because `evalJS()` will fail if page hasn't finished loading.
- Updated `evalJS()` documentation to state that it'll only work after the "load" event.
- Added WebView unit test "html-interop".

**Test:**
1. Set up "tiapp.xml" property "run-on-main-thread" to true.
2. Build and run "WebViewInteropTest.js" attached to [TIMOB-25859](https://jira.appcelerator.org/browse/TIMOB-25859) on Android.
3. Verify that you see a countdown in the WebView, starting from 5.
4. Verify that a toast message appears when it counts to zero, the web page reloads, and the countdown resets back to 5.
5. Look at the Android log and verify that no `Timeout waiting to evaluate JS` warning messages appear.
